externs/browser/html5.js

```diff
@@ -3180,6 +3180,121 @@ History.prototype.scrollRestoration;
  */
 Window.prototype.history;
 
+/**
+ * @interface
```

Comment:
FYI, this needs to be a constructor because it's an actual property of `window` and we need to be able to write `x instanceof Location`.

I will make the change internally to avoid the additional roundtrip.